File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {//[2,4,1]
        int buy=prices[0],lav=0;
        for(int i=1;i<prices.size();i++){
            int cost=prices[i]-buy;
            lav=max(lav,cost);
            buy=min(buy,prices[i]);
        }
        return lav;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int lav=0;
        int buy=Integer.MAX_VALUE;
        int sell=0;
        int buy_idx=0;

        for(int i=0;i<prices.length;i++){
            if(prices[i]<buy){
            buy=prices[i];
            buy_idx=i;
            sell=0;// for new buy prev sell not valid
            System.out.println("buy-"+buy);
            }
            if(prices[i]>sell && i>buy_idx){
            sell=prices[i];
            System.out.println("sell-"+sell);
            }
            lav=Math.max(lav,sell-buy);
        }
        return lav;
    }
}